pcomfortcloud/session.py

```diff
@@ -1,456 +1,425 @@
-'''
-Panasonic session, using Panasonic Comfort Cloud app api
-'''
-
+import base64
+import datetime
+import hashlib
 import json
+import random
+import string
+import time
+import urllib
 import requests
-import os
-import urllib3
-import hashlib
-
-from . import urls
-from . import constants
 
-def _validate_response(response):
-    """ Verify that response is OK """
-    if response.status_code == 200:
-        return
-    raise ResponseError(response.status_code, response.text)
+from bs4 import BeautifulSoup
+from . import exceptions
 
 
-class Error(Exception):
-    ''' Panasonic session error '''
-    pass
+def generate_random_string(length):
+    return ''.join(random.choice(string.ascii_letters + string.digits) for _ in range(length))
 
-class RequestError(Error):
-    ''' Wrapped requests.exceptions.RequestException '''
-    pass
 
+def generate_random_string_hex(length):
+    return ''.join(random.choice(string.hexdigits) for _ in range(length))
 
-class LoginError(Error):
-    ''' Login failed '''
-    pass
 
-class ResponseError(Error):
-    ''' Unexcpected response '''
-    def __init__(self, status_code, text):
-        super(ResponseError, self).__init__(
-            'Invalid response'
-            ', status code: {0} - Data: {1}'.format(
-                status_code,
-                text))
-        self.status_code = status_code
-        self.text = text
+def check_response(response, function_description, expected_status):
+    if response.status_code != expected_status:
+        raise exceptions.ResponseError(
+            f"({function_description}: Expected status code {expected_status}, received: {response.status_code}: " +
+            f"{response.text}"
+        )
 
 
-class Session(object):
-    """ Verisure app session
+def get_querystring_parameter_from_header_entry_url(response, header_entry, querystring_parameter):
+    header_entry_value = response.headers[header_entry]
+    parsed_url = urllib.parse.urlparse(header_entry_value)
+    params = urllib.parse.parse_qs(parsed_url.query)
+    return params.get(querystring_parameter, [None])[0]
 
-    Args:
-        username (str): Username used to login to verisure app
-        password (str): Password used to login to verisure app
 
-    """
+class PanasonicSession():
```

Comment:
Why the rename? And why the complete rewrite? Why can't PanasonicSession use ApiClient and still expose the methods it has, but using the ApiClient instead?